exec/java-exec/src/main/java/org/apache/drill/exec/server/options/TypeValidators.java

```diff
@@ -179,6 +181,7 @@ public void validate(final OptionValue v) {
 
     public TypeValidator(final String name, final Kind kind, final OptionValue defValue) {
       super(name);
+      checkArgument(defValue.type == OptionType.SYSTEM, "Default value must be SYSTEM type.");
```

Comment:
I am confused.. what does "these" refer to? I do not see any other precondition checks.

That change was addressing Hakim's comment [here](https://github.com/apache/drill/pull/159/files#diff-767c81796fcc968c436bd67655218f1aR83). The issue was when ControlsOptionValidator was being registered with the SystemOptionManager, the default option was being stored within SystemOptionManager as a SESSION option.